src/components/Dialogs/DialogManager.js

```diff
@@ -31,11 +33,13 @@ const DialogManager = () => {
     if (isDynamicDimension(dimension?.dimensionType)) {
         return <DynamicDimension dimension={dimension} onClose={onClose} />
     }
+    if (dimension?.dimensionType === DIMENSION_TYPE_PERIOD) {
```

Comment:
The logic in this file is starting to get pretty messy. How about having one main switch: 
```
    if (!dimension?.id) {
        return null
    }

    switch (dimension.dimensionType) {
        case DIMENSION_TYPE_CATEGORY:
        case DIMENSION_TYPE_CATEGORY_OPTION_GROUP_SET:
        case DIMENSION_TYPE_ORGANISATION_UNIT_GROUP_SET: {
            return <DynamicDimension dimension={dimension} onClose={onClose} />
        }
        case DIMENSION_TYPE_PERIOD: {
            return <PeriodDimension dimension={dimension} onClose={onClose} />
        }
        case DIMENSION_ID_ORGUNIT: {
            return <FixedDimension dimension={dimension} onClose={onClose} />
        }
        default: {
            return <ConditionsManager dimension={dimension} onClose={onClose} />
        }
    }
```